swan-lake/learn/running-ballerina-code.md

```diff
@@ -54,139 +47,92 @@ You can compile a source file with an entry point into an executable jar.
+`Config.toml` file may or may not provide configuration values for configurable variables that are used in the program. 
+Ballerina currently supports configurable variables of types `int`, `float`, `boolean`, `string`, `decimal` and the arrays of respective types. 
+See [Configurable BBE](/swan-lake/learn/by-example/configurable.html) for more details.
 
-### Sourcing Parameters into Ballerina Programs
-Configuration parameters for your programs and apps can be defined on the CLI, as an environment variable, or from a configuration file, with loading and override precedence in the same order.
-
-#### Sourcing CLI Parameters
-Consider the following example, which reads a Ballerina config value and prints it.
+Consider the following example, which uses configurable variables.
 
 ```ballerina
 import ballerina/io;
-import ballerina/config;
+
+configurable int id = ?;
+configurable string name = "Ann";
+configurable boolean married = true;
 
 public function main() {
-  string name = config:getAsString("hello.user.name");
-  io:println("Hello, " + name + " !");
+  io:println("User ID : ", id);
+  io:println("User Name : ", name);
+  io:println("Married : ", married);
 }
 ```
 
-The config key is `hello.user.name`. To pass a value to this config from the CLI, we can use `--key=value` format as the following command.
-```bash
-$ bal run  main.bal --hello.user.name=Ballerina
-Hello, Ballerina !
-```
-
-#### Sourcing Configuration Values
-
-The value can be passed as a config file as well. A configuration file should conform to the [TOML](https://github.com/toml-lang/toml) format. Ballerina only supports the following features of TOML: value types (string, int, float, and boolean), tables, and nested tables. Given below is a sample `ballerina.conf`:
-
+`?` denotes that `id` is a required configuration, hence `Config.toml` file must contain a value for key `id`.
+If a default value assigned, the configuration is optional, hence `Config.toml` file may or may not contain a values for
+configurable variables `name` and `married`.
+Consider the below `Config.toml` file.
 ```toml
-[hello.user]
-name="Ballerina"
+id = 1001
+name = "Jhone"
 ```
+ Since `Config.toml` file contains a value for key `name`, the program default
+value will be overridden by the value in the `Config.tomal` file.
 
-When running a program with config API lookups, Ballerina looks for a `ballerina.conf` file in the directory where the source files are located.
-
-If `ballerina.conf` resides in the same directory as `main.bal`, `balllerina run` can be used without any argument.
 ```bash
 $ bal run main.bal
-Hello, Ballerina !
-```
-To explicitly specify a configuration file, use the `--b7a.config.file` property. The path to the configuration file can be either an absolute or a relative path. 
-```bash
-$ bal run main.bal --b7a.config.file=path/to/conf/file/custom-config-file-name.conf
-Hello, Ballerina !
-```
-
-### Configuring Secrets as Configuration Items
```

Comment:
Need to add this content after fixing https://github.com/ballerina-platform/ballerina-lang/issues/28067